packages/arb-bridge-peripherals/contracts/tokenbridge/test/TestERC20.sol

```diff
@@ -29,3 +30,47 @@ contract TestERC20 is aeERC20 {
         _mint(msg.sender, 50000000);
     }
 }
+
+// test token code inspired from maker
+contract Bytes32ERC20 {
+    mapping(address => uint256) public balanceOf;
+    mapping(address => mapping(address => uint256)) public allowance;
+
+    function transfer(address dst, uint256 wad) public returns (bool) {
+        return transferFrom(msg.sender, dst, wad);
+    }
+
+    function transferFrom(
+        address src,
+        address dst,
+        uint256 wad
+    ) public returns (bool) {
+        if (src != msg.sender) {
+            allowance[src][msg.sender] = allowance[src][msg.sender] - wad;
+        }
+
+        balanceOf[src] = balanceOf[src] - wad;
+        balanceOf[dst] = balanceOf[dst] + wad;
+
+        return true;
+    }
+
+    function approve(address guy, uint256 wad) public returns (bool) {
+        allowance[msg.sender][guy] = wad;
+        return true;
+    }
+
+    function mint() public {
+        balanceOf[msg.sender] += 1 ether;
+    }
+}
+
+contract Bytes32ERC20WithMetadata is Bytes32ERC20 {
+    bytes32 public name = 0x4d616b6572000000000000000000000000000000000000000000000000000000;
+    bytes32 public symbol = 0x4d4b520000000000000000000000000000000000000000000000000000000000;
+    // TODO: what if this overflows?
+    uint8 public decimals = 18;
+
+    // no totalSupply field
+    // uint256 public totalSupply;
+}
```

Comment:
For testing purposes might be good to have these set by a constructor so that we can ex. add an all 0s case